processing/src/main/java/org/apache/druid/segment/transform/ExpressionTransform.java

```diff
@@ -90,7 +92,11 @@ private static Object getValueFromRow(final Row row, final String column)
     if (column.equals(ColumnHolder.TIME_COLUMN_NAME)) {
       return row.getTimestampFromEpoch();
     } else {
-      return row.getRaw(column);
+      Object raw = row.getRaw(column);
+      if (raw instanceof List) {
+        return ExpressionSelectors.coerceListDimToStringArray((List) raw);
```

Comment:
Is it fair to assume that all Lists are Lists of Strings?

What if the expression selector returns a long array?

Reply:
Since the only multi-value columns we support are strings it is currently a safe assumption to make. I believe `ExpressionTransform` would either need an `ValueType` for it's input type, or it need to be inferred from the input data, for us to assume anything else. Would you like me to add a `ValueType` parameter to this method at least that we can hard code to string for now?

Reply:
This isn't a multi-value column, though, is it? It looks like it's something from an input row and it could be an array of anything. I was thinking that since expressions support numeric arrays, it'd be good to keep them that way if found in the input rows.

Reply:
Fair, I was thinking backwards when I wrote this comment. I'll see what I can do, the tricky bit seems to be converting the list to the appropriate array type.